components/callbacks/config.go

```diff
@@ -66,3 +68,62 @@ func ConfigProvider(dc *dynamicconfig.Collection) *Config {
 		},
 	}
 }
+
+var AllowedAddresses = dynamicconfig.NewNamespaceTypedSettingWithConverter(
+	"component.callbacks.allowedAddresses",
+	allowedAddressConverter,
+	[]AddressMatchRule(nil),
+	`The per-namespace list of addresses that are allowed for callbacks and whether secure connections (https) are required.
+URLs are checked against each in order when starting a workflow with attached callbacks and only need to match one to pass validation.
+Default is no address rules, meaning all callbacks will be rejected. Any invalid entries are ignored. Each entry is a map with possible values:
+	 - "Pattern":string (required) the host:port pattern to which this config applies.
+		Wildcards, '*', are supported and can match any number of characters (e.g. '*' matches everything, 'prefix.*.domain' matches 'prefix.a.domain' as well as 'prefix.a.b.domain').
+	 - "AllowInsecure":bool (optional, default=false) indicates whether https is required`)
+
+type AddressMatchRule struct {
+	Regexp        *regexp.Regexp
```

Comment:
Interesting! I hadn't considered compiling things. I see two problems here:
- The conversion function is called every time you get a value from the dc, in that case that looks like it'll be every StartWorkflowExecution (that has callbacks). I think it should probably cache things better for cpu and also to avoid garbage from compiling the regexes over and over.
- The subscription stuff uses reflect.DeepEqual to compare the post-converted values, so I assume two converted values will never compare equal, which will throw that off. If you never subscribe to this key that's probably okay. (But should add a comment about that.)